src/libs/antares/study/study.cpp

```diff
@@ -224,284 +224,25 @@ uint64 Study::memoryUsage() const
-        if (setSize && (setSize < minNbYearsInParallel))
-            minNbYearsInParallel = setSize;
-    }
-
-    // GUI : storing minimum number of parallel years (in a set of parallel years).
-    //		 Useful in the run window's simulation cores field in case parallel mode is enabled
-    // by user.
-    minNbYearsInParallel_save = minNbYearsInParallel;
-
-    // The max number of years to run in parallel is limited by the max number years in a set of
-    // parallel years. This latter number can be limited by the smallest interval between 2 refresh
-    // points and determined by the unrun MC years in case of play-list.
-    uint maxNbYearsOverAllSets = 0;
-    for (uint s = 0; s < setsOfParallelYears.size(); s++)
-    {
-        if (setsOfParallelYears[s].size() > maxNbYearsOverAllSets)
-            maxNbYearsOverAllSets = (uint)setsOfParallelYears[s].size();
-    }
-    maxNbYearsInParallel = maxNbYearsOverAllSets;
-
-    // GUI : storing max nb of parallel years (in a set of parallel years) in case parallel mode is
-    // enabled.
-    //		 Useful for RAM estimation.
-    maxNbYearsInParallel_save = maxNbYearsInParallel;
-
-    // Here we answer the question (useful only if hydro hot start is asked) : do all sets of
-    // parallel years have the same size ?
-    if (parameters.initialReservoirLevels.iniLevels == Antares::Data::irlHotStart
-        && setsOfParallelYears.size() && maxNbYearsInParallel > 1)
-    {
-        uint currentSetSize = (uint)setsOfParallelYears[0].size();
-        if (setsOfParallelYears.size() > 1)
-        {
-            for (uint s = 1; s < setsOfParallelYears.size(); s++)
-            {
-                if (setsOfParallelYears[s].size() != currentSetSize)
-                {
-                    parameters.allSetsHaveSameSize = false;
-                    break;
-                }
-            }
-        }
-    } // End if hot start
+void Study::getNumberOfCores(const bool forceParallel, const bool enableParallel, const uint nbYearsParallelForced) {
+
+    TempAreaListHolder holder{};
+    bool thermalTSRefresh = holder.checkThermalTSGeneration(folderInput);
+
+    SetsOfParallelYearCalculator setsBuilder(forceParallel,
+                                             enableParallel,
+                                             nbYearsParallelForced, Yuni::System::CPU::Count(),
+                                             thermalTSRefresh,
+                                             parameters);
+    // For GUI
+    minNbYearsInParallel_save = setsBuilder.getMinNbParallelYearsForGUI();
+    maxNbYearsInParallel_save = setsBuilder.getForcedNbOfParallelYears();
+
+    // For the solver
+    maxNbYearsInParallel = setsBuilder.getForcedNbOfParallelYears();
+    parameters.allSetsHaveSameSize = setsBuilder.allSetsParallelYearsHaveSameSize();
+    setsOfParallelYears = setsBuilder.getSetsOfParallelYears();
+    pNbYearsReallyPerformed = setsBuilder.getNbYearsReallyPerformed();
```

Comment:
About : 
```C++
pNbYearsReallyPerformed = setsBuilder.getNbYearsReallyPerformed();
```
If we agree that computing the number of performed years in the study is not part of the responsibility of the parallel years orchestrator, this line should be removed.